app/assets/javascripts/components/story/ExpandedStory/ExpandedStoryLabels.js

```diff
@@ -6,51 +6,49 @@ class ExpandedStoryLabels extends React.Component {
   constructor(props) {
     super(props);
 
-    this.state = {
-      tags: props.labels
-    }
-
     this.handleAddition = this.handleAddition.bind(this);
     this.handleDelete = this.handleDelete.bind(this);
   }
 
   handleDelete(index) {
-    const { onEdit } = this.props;
-    const tags = this.state.tags.filter((tag, tagIndex) => tagIndex !== index);
+    const { onEdit, labels } = this.props;
 
-    this.setState(
-      { tags }, () => {
-        onEdit(this.state.tags)
-      }
+    onEdit(labels.filter(
```

Comment:
It would be good to not do this here. We could have a prop called `onRemoveLabel` or something like that and leave it to the reducer to do the filter 😄